In [4]:
import pandas as pd
import numpy as np

In [5]:
df = pd.read_excel('File 1.xlsx', sheet_name='covid_19_data')
df.head()

,SNo,ObservationDate,Province/State,Country/Region,Last Update,Confirmed,Deaths,Recovered
0,1,01/22/2020,Anhui,Mainland China,1/22/2020 17:00,1,0,0
1,2,01/22/2020,Beijing,Mainland China,1/22/2020 17:00,14,0,0
2,3,01/22/2020,Chongqing,Mainland China,1/22/2020 17:00,6,0,0
3,4,01/22/2020,Fujian,Mainland China,1/22/2020 17:00,1,0,0
4,5,01/22/2020,Gansu,Mainland China,1/22/2020 17:00,0,0,0


In [7]:
# Check for null values and basic info
df.info(), df.isnull().sum(), df.describe(), df.shape, df.dtypes, df.columns, df.head(), df.value_counts()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 205951 entries, 0 to 205950
Data columns (total 8 columns):
 #   Column           Non-Null Count   Dtype 
---  ------           --------------   ----- 
 0   SNo              205951 non-null  int64 
 1   ObservationDate  205951 non-null  object
 2   Province/State   150571 non-null  object
 3   Country/Region   205951 non-null  object
 4   Last Update      205951 non-null  object
 5   Confirmed        205951 non-null  int64 
 6   Deaths           205951 non-null  int64 
 7   Recovered        205951 non-null  int64 
dtypes: int64(4), object(4)
memory usage: 12.6+ MB


(None,
 SNo                    0
 ObservationDate        0
 Province/State     55380
 Country/Region         0
 Last Update            0
 Confirmed              0
 Deaths                 0
 Recovered              0
 dtype: int64,
                  SNo     Confirmed         Deaths     Recovered
 count  205951.000000  2.059510e+05  205951.000000  2.059510e+05
 mean   102976.000000  4.540148e+04    1262.633850  2.766348e+04
 std     59453.076982  1.437216e+05    4082.838603  1.334517e+05
 min         1.000000 -3.028440e+05    -178.000000 -8.544050e+05
 25%     51488.500000  5.960000e+02       7.000000  1.100000e+01
 50%    102976.000000  5.361000e+03     101.000000  1.028000e+03
 75%    154463.500000  2.618400e+04     713.000000  9.880000e+03
 max    205951.000000  3.049037e+06   80143.000000  6.399531e+06,
 (205951, 8),
 SNo                 int64
 ObservationDate    object
 Province/State     object
 Country/Region     object
 Last Update        object
 Confirmed           int64
 Deaths 

# 🧼 Data Cleaning Summary
✅ Key Points:
* Total rows: 205,951
* Columns: 8
* Column Province/State has ~55,000 missing values → that’s expected for countries without provinces (e.g., Italy, Iran).
* All other columns are clean and complete.
* Dates (ObservationDate) are stored as strings, so we’ll convert them to datetime for proper analysis.

In [9]:
# Convert date column to datetime
df['ObservationDate'] = pd.to_datetime(df['ObservationDate'])

# Group by Date and Country, summing confirmed, deaths, and recovered
grouped_df = df.groupby(['ObservationDate', 'Country/Region'], as_index=False)[['Confirmed', 'Deaths', 'Recovered']].sum()

# Save the cleaned and grouped data for Tableau
cleaned_path = "covid_cleaned_for_tableau.csv"
grouped_df.to_csv(cleaned_path, index=False)

# Show the first few rows of the cleaned dataset
print(grouped_df.head())

  ObservationDate Country/Region  Confirmed  Deaths  Recovered
0      2020-01-02      Australia         12       0          2
1      2020-01-02       Cambodia          1       0          0
2      2020-01-02         Canada          4       0          0
3      2020-01-02        Finland          1       0          0
4      2020-01-02         France          6       0          0
